### ISSUE TITLE
Feature: Skip silence (for PREVIEW MODE)

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#1231]
I implemented a "skip initial silence" mode before, and removed it, because I didn't really use it.  BUT, I think I would use it every time for PREVIEW MODE (clicking on the little play button on the L hand side of a line in the songTable).

I just don't want to wait for 10 seconds, especially on these older songs, or songs that were digitized from 45RPM records to MP3, for the song to start.  When I want a preview, I want it right away.

I am wondering if I can get away with NO preference for this, just always skip the silence.  I think that might work OK.

And if we still want to skip silence in regular playback, I'm still thinking that we can use markers (partially implemented already), or I can special case the Start Loop code, to create a Start Song marker (as Mixxx does).  That could be set on a song-by-song basis, and a user could tune the length to be whatever he wants, OR it could even start after the intro section of a song, if it's particularly long.

**Comment by pkbott** [comment]
I like this auto skip as well. Just curious though since I don’t use dark mode, is it crazy intense to make the new light mode?  Would love the preview feature for sure. Ready to try something FUN, UNIQUE and ***@***.*** Square Dance Group:  https://www.facebook.com/groups/squaredanceforfun/On Dec 14, 2024, at 4:22 AM, Mike Pogue ***@***.***> wrote:﻿
I implemented a "skip initial silence" mode before, and removed it, because I didn't really use it.  BUT, I think I would use it every time for PREVIEW MODE (clicking on the little play button on the L hand side of a line in the songTable).
I just don't want to wait for 10 seconds, especially on these older songs, or songs that were digitized from 45RPM records to MP3, for the song to start.  When I want a preview, I want it right away.
I am wondering if I can get away with NO preference for this, just always skip the silence.  I think that might work OK.
And if we still want to skip silence in regular playback, I'm still thinking that we can use markers (partially implemented already), or I can special case the Start Loop code, to create a Start Song marker (as Mixxx does).  That could be set on a song-by-song basis, and a user could tune the length to be whatever he wants, OR it could even start after the intro section of a song, if it's particularly long.

**Comment by mpogue2** [comment]
It's not crazy intense, it just requires some artistic ability and available time to design the light mode so that it looks good, and then time to implement it.  It's probably 2 hours of design time, and 4-8 hours of implementation (and testing) time (depending on how complicated it gets, and what bugs it uncovers).

**Comment by mpogue2** [comment]
@pkbott Please see #930 .  I am starting to look at what it would take to do the new Light Mode.  It's probably a lot more than 4-8 hours of implementation, but it will provide some pretty serious customization ability when it's available.  See #930 to follow along at home as I play with this!

**Comment by mpogue2** [comment]
This feature is relatively difficult to implement with the code as it is today.  Right now, pressing an audition/preview button just creates a music player, hands that player a pathname to a music file, and tells it to play.  

While there is a way to tell the Player to seek to a position (if (isSeekable() setPosition()), there is no way to know where the end of the initial silence is, unless we have decoded the file at some time in the past, and we have cached that value.

**Comment by mpogue2** [comment]
Maybe the start marker could be calculated on first load, with a preference: "[X] Audition playback skips initial silence".

Or even simpler, we could just do: "Audition starts at [ ] seconds" instead, or as a temporary improvement.

**Comment by mpogue2** [comment]
Or, [X] Audition starts at loop IN point .
I think this would work well for both patter and singers, and it would not require markers.

**Comment by pkbott** [comment]
I agree this would be good. Ready to try something FUN, UNIQUE and ***@***.*** Square Dance Group:  https://www.facebook.com/groups/squaredanceforfun/On Jul 6, 2025, at 1:35 AM, Mike Pogue ***@***.***> wrote:﻿mpogue2 left a comment (mpogue2/SquareDesk#1231)
Or, [X] Audition starts at loop IN point .
I think this would work well for both patter and singers, and it would not require markers.

**Comment by mpogue2** [comment]
The audition button or the keyboard shortcut (OPTION-/) will now start playback **approximately** 1 measure before Patter (start loop point) and Singing call (start of Figure 1 point).   That provides the user's ear with some context for the song, and it also guarantees that we are way past the initial silence.  This should work well, because in general all patters have loop points set, and all singers have the Intro and Outro points set, so all songs should automatically skip silence.  If no loop point have been set, then audition defaults back to starting at the beginning of the song.

NOTE: Because the measure finder algorithm can be up to one beat off from the audible measure start (depending on the song), the user might actually hear 4, 5, or 6 beats of music before the loop/figure 1 point.  Whether this happens depends on whether the user used Snap To Beat (which is always accurate) or Snap To Measure (which can be one beat off sometimes).  In practice being off-by-one does not hurt the loop, because it's off at start loop and end loop positions by the same amount, and therefore the loop is still seamless.  But, the user might notice that the Audition/preview is not always the same number of beats.

Commit: f2922d7b40ac27cdcc8d083735fe02cffb0566c9
